### PR TITLE
Clean BBReferenceProcessor code

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Refactoring BBReferenceProcessor to handle single and multiple references separately. Now, both systems have different data shapes (single id vs array), so having the code all together was keeping it hard to follow

### CODE REVIEW

File: packages/server/src/utilities/rowProcessor/bbReferenceProcessor.ts [hunk 1]
```diff
@@ -10,167 +9,124 @@ import { InvalidBBRefError } from "./errors"
...
+export async function processInputBBReference(
   value: string | { _id: string },
-  type: FieldType.BB_REFERENCE_SINGLE
-): Promise<string | null>
-export function processInputBBReferences(
-  value: string | string[] | { _id: string } | { _id: string }[],
-  type: FieldType.BB_REFERENCE,
-  subtype: BBReferenceFieldSubType
-): Promise<string | null>
+  subtype: BBReferenceFieldSubType.USER
```
[the remark this answers is not in the document]
Comment (samwho) [reply packages/server/src/utilities/rowProcessor/bbReferenceProcessor.ts:14]: Ahh yes, that makes a lot of sense. Good thinking!

File: packages/server/src/utilities/rowProcessor/bbReferenceProcessor.ts [hunk 2]
```diff
@@ -10,167 +9,124 @@ import { InvalidBBRefError } from "./errors"
...
-  type: FieldType.BB_REFERENCE,
-  subtype: BBReferenceFieldSubType
-): Promise<UserReferenceInfo[] | undefined>
-
-export async function processOutputBBReferences(
-  value: string | string[],
-  type: FieldType.BB_REFERENCE | FieldType.BB_REFERENCE_SINGLE,
-  subtype?: BBReferenceFieldSubType
-) {
+  subtype: BBReferenceFieldSubType.USER
```
[the remark this answers is not in the document]
Comment (adrinr) [reply packages/server/src/utilities/rowProcessor/bbReferenceProcessor.ts:112]: Same than before

**Review by samwho — APPROVED** [review]
